fs/rails-base-graphql-api
# Update user avatar with uploaded data

### Summary

This PR adds ability to update User model with uploaded image data

### How it works

![Screen Shot 2020-06-22 at 19 41 26](https://user-images.githubusercontent.com/58729845/85313201-85384780-b4c0-11ea-8a16-87fa1c0c928d.png)

![Screen Shot 2020-06-24 at 16 26 13](https://user-images.githubusercontent.com/58729845/85566681-ee42cb00-b638-11ea-85c7-9223644aa4ba.png)

### Test plan

List of steps to manually test introduced functionality:

* Upload image to S3 with Uppy or manually (to 'cache' directory)
* Open GraphiQL App
* Make request using schema:
```
mutation SignUp($email: String!, $password: String!, $firstName: String!, $lastName: String!, $avatar: ImageUploader!) {
  signup(email: $email, password: $password, firstName: $firstName, lastName: $lastName, avatar: $avatar) {
    me {
      id
      email
      firstName
      lastName
      avatarUrl
    }
    accessToken
    refreshToken
  }
}

```
* Check that image has uploaded to 'storage' directory
* The response should content avatar URL

### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### References
* Resolves #42
